infra/modules/services/diagnosticsettings.bicep

```diff
@@ -0,0 +1,304 @@
+  parent: synapseworkspace
+  name: sparkPool
+}]
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' existing = {
+  name: logAnalyticsName
+}
+
+resource cosmosdb 'Microsoft.DocumentDB/databaseAccounts@2021-03-15' existing = {
+  name: cosmosdbName
+}
+
+resource iothub 'Microsoft.Devices/IotHubs@2021-03-31' existing = {
+  name: iothubName
+}
+
+resource sqlserver 'Microsoft.Sql/servers@2020-11-01-preview' existing = {
+  name: sqlserverName
+}
+
+resource sqlserverDatabase001 'Microsoft.Sql/servers/databases@2020-11-01-preview' existing = {
+  parent: sqlserver
+  name: 'Database001'
+}
+
+resource eventhubNamespace 'Microsoft.EventHub/namespaces@2021-06-01-preview' existing = {
+  name: eventhubnamespaceName
+}
+
+resource streamanalyticsjob001 'Microsoft.StreamAnalytics/streamingjobs@2020-03-01' existing = {
+  name: streamanalyticsName
+}
+
+resource diagnosticSetting001 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  scope: synapseworkspace
+  name: 'diagnostic-${synapseworkspace.name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'SynapseRbacOperations'
+        enabled: true
+      }
+      {
+        category: 'GatewayApiRequests'
+        enabled: true
+      }
+      {
+        category: 'BuiltinSqlReqsEnded'
+        enabled: true
+      }
+      {
+        category: 'IntegrationPipelineRuns'
+        enabled: true
+      }
+      {
+        category: 'IntegrationActivityRuns'
+        enabled: true
+      }
+      {
+        category: 'IntegrationTriggerRuns'
+        enabled: true
+      }
+    ]
+  }
+}
+
+resource diagnosticSetting2 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for i in range(0, synapseSqlPoolsCount): {
+  scope: synapsesqlpool[i]
+  name: 'diagnostic-${synapseworkspace.name}-${synapsesqlpool[i].name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'SqlRequests'
+        enabled: true
+      }
+      {
+        category: 'RequestSteps'
+        enabled: true
+      }
+      {
+        category: 'ExecRequests'
+        enabled: true
+      }
+      {
+        category: 'DmsWorkers'
+        enabled: true
+      }
+      {
+        category: 'Waits'
+        enabled: true
+      }
+    ]
+  }
+}]
+
+resource diagnosticSetting3 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for i in range(0, synapseSparkPoolCount): {
+  scope: synapsebigdatapool[i]
+  name: 'diagnostic-${synapseworkspace.name}-${synapsebigdatapool[i].name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'BigDataPoolAppsEnded'
+        enabled: true
+      }
+    ]
+  }
+}]
+
+resource diagnosticSetting4 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  scope: cosmosdb
```

Comment:
Will this work if the  cosmos id set to false in params ?

Reply:
I get: 
{
  'code': 'InvalidTemplate',
  'message': "Deployment template validation failed: 'The template resource 'Microsoft.Resources/deployments/dashboard' reference      to 'Microsoft.Resources/deployments/cosmos001' requires an API version. Please see https://aka.ms/arm-template for usage details.'.",
  'additionalInfo': [
    {
      'type': 'TemplateViolation',
      'info': {
        'lineNumber': 0,
        'linePosition': 0,
        'path': ''
      }
    }
  ]
}